### DISCUSSION TITLE
TLS_EXT_SERVER_NAME in libtls

### DISCUSSION THREAD

**Discussion by ghost** [strongswan/strongswan#410] [Q&A] [answered]
is it possible to send a customized server name in send client hello? how?

**Comment by tobiasbrunner** [comment, marked as answer]
In what context? EAP? If so, did you try setting the AAA identity to a FQDN?

**Reply by ghost** [reply]
yes EAP. it seems send_client_hello is never called in tls_peer.

**Reply by ghost** [reply]
> In what context? EAP? If so, did you try setting the AAA identity to a FQDN?

how to set the AAA identity? is this for DNS?

**Reply by tobiasbrunner** [reply]
Please provide configs and logs.

**Reply by tobiasbrunner** [reply]
> how to set the AAA identity? is this for DNS?

In the config (refer to the man or wiki pages). But if `send_client_hello()` is never called you probably have to fix other problems first.

**Reply by ghost** [reply]
this is my config

[removed]

**Reply by ghost** [reply]
it connects and worked well, I want to have a customized server name if its possible?

**Reply by tobiasbrunner** [reply]
Ah, you are talking about the Android client. There is currently no setting for the AAA identity. But you could try setting _remote.id_. However, that also changes the remote IKE identity, so this won't work if the two identities are different.

**Reply by tobiasbrunner** [reply]
> have a customized server name

What does that mean?

**Reply by ghost** [reply]
> > have a customized server name
> 
> What does that mean?

basically overrides the tls server name indication

**Reply by ghost** [reply]
I tried to set server identity, it says auth failed.

**Reply by tobiasbrunner** [reply]
But you say it already works. So what do you mean with "customized server name"? And to what purpose?

**Reply by ghost** [reply]
its for bypassing a network filter, usually works with my apps.

**Reply by tobiasbrunner** [reply]
Absolutely irrelevant here as the EAP authenticaiton is encrypted in IKEv2.

**Reply by ghost** [reply]
an example https://security.stackexchange.com/questions/189591/is-there-any-way-to-bypass-sni-based-https-filtering

**Reply by tobiasbrunner** [reply]
Again, this is completely irrelevant as strongSwan does not use TLS except for EAP methods that are exchanged encrypted within IKEv2 (the actual protocol used to create the VPN).

**Reply by ghost** [reply]
ok understood. thanks!
